code/datums/manufacturing_requirements.dm

```diff
@@ -1,216 +1,336 @@
-	proc/matches_property(var/datum/material/M)
-		return M.getProperty(src.material_property) >= src.material_threshold
+	/// Returns whether the material property meets the threshold. Overwrite to have behavior other than >=
+	proc/match_property(var/datum/material/M)
+		return M.getProperty(src.property_id) >= src.property_threshold
 
 /datum/manufacturing_requirement/match_property/conductive
 	name = "Conductive"
 	id = "conductive"
-	material_property = "electrical"
-	material_threshold = 6
+	property_id = "electrical"
+	property_threshold = 6
 
 /datum/manufacturing_requirement/match_property/conductive/high
 	name = "High Energy Conductor"
 	id = "conductive_high"
-	material_threshold = 8
-
-/datum/manufacturing_requirement/match_property/crystal
-	name = "Crystal"
-	id = "crystal"
-	material_flags = MATERIAL_CRYSTAL
-
-/datum/manufacturing_requirement/match_property/crystal/dense
-	name = "Extraordinarily Dense Crystalline Matter"
-	id = "crystal_dense"
-	material_property = "density"
-	material_threshold = 7
-
-/datum/manufacturing_requirement/match_property/crystal/gemstone
-	name = "Gemstone"
-	id = "gemstone"
-
-	is_match(var/datum/material/M)
-		if (!(istype(M, /datum/material/crystal/gemstone)))
-			return FALSE
-		. = ..()
+	property_threshold = 8
 
 /datum/manufacturing_requirement/match_property/dense
 	name = "High Density Matter"
 	id = "dense"
-	material_property = "density"
-	material_threshold = 4
+	property_id = "density"
+	property_threshold = 4
 
-/datum/manufacturing_requirement/match_property/dense/super
+/datum/manufacturing_requirement/match_property/superdense
 	name = "Very High Density Matter"
 	id = "dense_super"
-	material_threshold = 6
+	property_id = "density"
+	property_threshold = 6
+
+/datum/manufacturing_requirement/match_property/ultradense
+	name = "Ultra-Dense Matter"
+	id = "dense_property_ultra"
+	property_id = "density"
+	property_threshold = 7
 
 /datum/manufacturing_requirement/match_property/energy
-	name = "Power Source"
-	id = "energy"
-	material_property = "radioactive"
-	material_flags = MATERIAL_ENERGY
+	name = "Radioactive"
+	id = "energy_property"
+	property_id = "radioactive"
+	property_threshold = 3
 
 /datum/manufacturing_requirement/match_property/energy/high
-	name = "Significant Power Source"
-	id = "energy_high"
-	material_threshold = 3
-
-/datum/manufacturing_requirement/match_property/energy/extreme
-	name = "Extreme Power Source"
-	id = "energy_extreme"
-	material_threshold = 5
-
-/datum/manufacturing_requirement/match_property/fabric
-	name = "Fabric"
-	id = "fabric"
-	material_flags = MATERIAL_CLOTH | MATERIAL_RUBBER | MATERIAL_ORGANIC
+	name = "Highly Radioactive"
+	id = "energy_property_high"
+	property_threshold = 5
 
 /datum/manufacturing_requirement/match_property/insulated
-	name = "Insulative"
-	id = "insulated"
-	material_flags = MATERIAL_CLOTH | MATERIAL_RUBBER
-	material_threshold = 4
+	name = "Insulated Material"
```

Comment:
```suggestion
	name = "Insulative Material"
```
The material itself is not insulated, *it* is the insulator and thus the thing you make out of it would be insulated.